superset/models/core.py

```diff
@@ -611,10 +611,11 @@ def alter_positions(dashboard, old_to_new_slc_id_dict):
 
         # override the dashboard
         existing_dashboard = None
+        remote_id = dashboard_to_import.params_dict.get("remote_id")
         for dash in session.query(Dashboard).all():
             if (
                 "remote_id" in dash.params_dict
-                and dash.params_dict["remote_id"] == dashboard_to_import.id
+                and dash.params_dict["remote_id"] == remote_id
```

Comment:
it would be nice to add a unit test

Reply:
Yes. Looked at the tests and seems to testing different approach that the one implemented but still I'm missing many parts.

This test should have cough the error: `test_import_override_dashboard_2_slices` but executes a different logic than the current one implemented.
Do you think we cab move on without the test? Ofc, is working on our project 😅 "Trust me I'm engineer!"

Reply:
I was working on this part of the codebase a while ago and this one is quite tricky and easy to introduce a bug. 
In addition to that if somebody would refactor this file in the future it would be nice to provide them some guardrails. This is not a blocker but just a suggestion.

A good test case would be importing:
dash1 {id: 1, remote_id: 101}
dash2 {id:2, remote_id: 101}
and making sure that override will happen